### PR TITLE
feat(iam): support eps role assignment in user and user group

### PR DESCRIPTION
**Pull request by Jason-Zhang9309**
**What this PR does / why we need it**:

**Which issue this PR fixes**:
*(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*
fixes #xxx

**Special notes for your reviewer**:

**Release note**:

```release-note
1. add new resource huaweicloud_identity_user_role_assignment
2. write huaweicloud_identity_role_assignment to huaweicloud_identity_group_role_assignment, the old resource is 
   still available, in other words, they are using the same code
```

## PR Checklist

* [x] Tests added/passed.
* [x] Documentation updated.
* [x] Schema updated.

## Acceptance Steps Performed

```
make testacc TEST='./huaweicloud/services/acceptance/iam' TESTARGS='-run TestAccIdentityUserRoleAssignment_basic'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud/services/acceptance/iam -v -run TestAccIdentityUserRoleAssignment_basic -timeout 360m -parallel 4
=== RUN   TestAccIdentityUserRoleAssignment_basic
=== PAUSE TestAccIdentityUserRoleAssignment_basic
=== CONT  TestAccIdentityUserRoleAssignment_basic
--- PASS: TestAccIdentityUserRoleAssignment_basic (29.88s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/iam       30.015s

make testacc TEST='./huaweicloud/services/acceptance/iam' TESTARGS='-run TestAccIdentityGroupRoleAssignment_basic'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud/services/acceptance/iam -v -run TestAccIdentityGroupRoleAssignment_basic -timeout 360m -parallel 4
=== RUN   TestAccIdentityGroupRoleAssignment_basic
=== PAUSE TestAccIdentityGroupRoleAssignment_basic
=== CONT  TestAccIdentityGroupRoleAssignment_basic
--- PASS: TestAccIdentityGroupRoleAssignment_basic (16.38s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/iam       16.473s

make testacc TEST='./huaweicloud/services/acceptance/iam' TESTARGS='-run TestAccIdentityGroupRoleAssignment_project'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud/services/acceptance/iam -v -run TestAccIdentityGroupRoleAssignment_project -timeout 360m -parallel 4
=== RUN   TestAccIdentityGroupRoleAssignment_project
=== PAUSE TestAccIdentityGroupRoleAssignment_project
=== CONT  TestAccIdentityGroupRoleAssignment_project
--- PASS: TestAccIdentityGroupRoleAssignment_project (14.40s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/iam       14.496s

make testacc TEST='./huaweicloud/services/acceptance/iam' TESTARGS='-run TestAccIdentityGroupRoleAssignment_epsID'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud/services/acceptance/iam -v -run TestAccIdentityGroupRoleAssignment_epsID -timeout 360m -parallel 4
=== RUN   TestAccIdentityGroupRoleAssignment_epsID
=== PAUSE TestAccIdentityGroupRoleAssignment_epsID
=== CONT  TestAccIdentityGroupRoleAssignment_epsID
--- PASS: TestAccIdentityGroupRoleAssignment_epsID (15.43s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/iam       15.510s
```

### CODE REVIEW

File: docs/resources/identity_group_role_assignment.md [hunk 1]
```diff
@@ -0,0 +1,123 @@
+---
+subcategory: "Identity and Access Management (IAM)"
+---
+
+# huaweicloud_identity_group_role_assignment
+
+Manages a Role assignment within group on HuaweiCloud IAM Service.
```
Comment (ShiChangkuo) [inline docs/resources/identity_group_role_assignment.md]:
```suggestion
Manages an IAM user group role assignment within HuaweiCloud IAM Service.
```

File: docs/resources/identity_group_role_assignment.md [hunk 2]
```diff
@@ -0,0 +1,123 @@
...
+
+## Attributes Reference
+
+In addition to all arguments above, the following attributes are exported:
+
+* `id` - The resource ID. When assign in domain, the format is `<group_id>/<role_id>/<domain_id>`;
+  when assign in project, the format is `<group_id>/<role_id>/<project_id>`;
+  when assign in enterprise project, the format is `<group_id>/<role_id>/<enterprise_project_id>`;
+
+The role assignments can be imported using the `group_id`, `role_id` and  `domain_id`, `project_id`,
```
Comment (ShiChangkuo) [inline docs/resources/identity_group_role_assignment.md:110]:
```suggestion
## Import

The role assignments can be imported using the `group_id`, `role_id` and  `domain_id`, `project_id`,
```

File: docs/resources/identity_user_role_assignment.md [hunk 3]
```diff
@@ -0,0 +1,57 @@
+---
+subcategory: "Identity and Access Management (IAM)"
+---
+
+# huaweicloud_identity_user_role_assignment
+
+Manages a IAM user role assignment within HuaweiCloud IAM.
```
Comment (ShiChangkuo) [inline docs/resources/identity_user_role_assignment.md]:
```suggestion
Manages an IAM user role assignment within HuaweiCloud IAM.
```

File: huaweicloud/services/acceptance/iam/resource_huaweicloud_identity_role_assignment_test.go
Comment (ShiChangkuo) [inline huaweicloud/services/acceptance/iam/resource_huaweicloud_identity_role_assignment_test.go:1]: can we keep this file for compatibility?

File: huaweicloud/services/acceptance/iam/resource_huaweicloud_identity_group_role_assignment_test.go [hunk 4]
```diff
@@ -0,0 +1,295 @@
...
+
+	if projectID != "" {
+		return iam.GetGroupRoleAssignmentWithProjectID(identityClient, groupID, roleID, projectID)
+	}
+
+	if enterpriseProjectID != "" {
+		return iam.GetGroupRoleAssignmentWithEpsID(iamClient, groupID, roleID, enterpriseProjectID)
+	}
+
+	return nil, nil
```
Comment (ShiChangkuo) [inline huaweicloud/services/acceptance/iam/resource_huaweicloud_identity_group_role_assignment_test.go]: shall we return an error here?

File: docs/resources/identity_group_role_assignment.md (hunk 1 above)
Comment (Jason-Zhang9309) [reply docs/resources/identity_group_role_assignment.md]: fixed

File: docs/resources/identity_group_role_assignment.md (hunk 2 above)
Comment (Jason-Zhang9309) [reply docs/resources/identity_group_role_assignment.md:110]: fixed

File: docs/resources/identity_user_role_assignment.md (hunk 3 above)
Comment (Jason-Zhang9309) [reply docs/resources/identity_user_role_assignment.md]: fixed

File: huaweicloud/services/acceptance/iam/resource_huaweicloud_identity_role_assignment_test.go
Comment (Jason-Zhang9309) [reply huaweicloud/services/acceptance/iam/resource_huaweicloud_identity_role_assignment_test.go:1]: due to the ID format difference, this test file will not work, but I can add a test case in the new test file, to check whether the old resource still works

File: huaweicloud/services/acceptance/iam/resource_huaweicloud_identity_group_role_assignment_test.go (hunk 4 above)
Comment (Jason-Zhang9309) [reply huaweicloud/services/acceptance/iam/resource_huaweicloud_identity_group_role_assignment_test.go]: fixed

**Comment by Jason-Zhang9309** [comment]
```
make testacc TEST='./huaweicloud/services/acceptance/iam' TESTARGS='-run TestAccIdentityUserRoleAssignment_basic'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud/services/acceptance/iam -v -run TestAccIdentityUserRoleAssignment_basic -timeout 360m -parallel 4
=== RUN   TestAccIdentityUserRoleAssignment_basic
=== PAUSE TestAccIdentityUserRoleAssignment_basic
=== CONT  TestAccIdentityUserRoleAssignment_basic
--- PASS: TestAccIdentityUserRoleAssignment_basic (8.31s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/iam       8.382s

make testacc TEST='./huaweicloud/services/acceptance/iam' TESTARGS='-run TestAccIdentityGroupRoleAssignment_basic'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud/services/acceptance/iam -v -run TestAccIdentityGroupRoleAssignment_basic -timeout 360m -parallel 4
=== RUN   TestAccIdentityGroupRoleAssignment_basic
=== PAUSE TestAccIdentityGroupRoleAssignment_basic
=== CONT  TestAccIdentityGroupRoleAssignment_basic
--- PASS: TestAccIdentityGroupRoleAssignment_basic (9.08s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/iam       9.178s

make testacc TEST='./huaweicloud/services/acceptance/iam' TESTARGS='-run TestAccIdentityGroupRoleAssignment_project'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud/services/acceptance/iam -v -run TestAccIdentityGroupRoleAssignment_project -timeout 360m -parallel 4
=== RUN   TestAccIdentityGroupRoleAssignment_project
=== PAUSE TestAccIdentityGroupRoleAssignment_project
=== CONT  TestAccIdentityGroupRoleAssignment_project
--- PASS: TestAccIdentityGroupRoleAssignment_project (9.62s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/iam       9.692s

make testacc TEST='./huaweicloud/services/acceptance/iam' TESTARGS='-run TestAccIdentityGroupRoleAssignment_epsID'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud/services/acceptance/iam -v -run TestAccIdentityGroupRoleAssignment_epsID -timeout 360m -parallel 4
=== RUN   TestAccIdentityGroupRoleAssignment_epsID
=== PAUSE TestAccIdentityGroupRoleAssignment_epsID
=== CONT  TestAccIdentityGroupRoleAssignment_epsID
--- PASS: TestAccIdentityGroupRoleAssignment_epsID (8.64s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/iam       8.702s

make testacc TEST='./huaweicloud/services/acceptance/iam' TESTARGS='-run TestAccIdentityGroupRoleAssignment_old'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud/services/acceptance/iam -v -run TestAccIdentityGroupRoleAssignment_old -timeout 360m -parallel 4
=== RUN   TestAccIdentityGroupRoleAssignment_old
=== PAUSE TestAccIdentityGroupRoleAssignment_old
=== CONT  TestAccIdentityGroupRoleAssignment_old
--- PASS: TestAccIdentityGroupRoleAssignment_old (9.28s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/iam       9.345s
```

**Pull request merged**